integrations/threejs/threejs.html

```diff
@@ -0,0 +1,295 @@
+        document.body.appendChild(WebGL.getWebGL2ErrorMessage());
+      }
+
+      THREE.Object3D.DEFAULT_UP.set(0, 0, 1);
+
+      let scene, mesh, renderer, stats;
+      let colourTextureGL, udTextureDepth;
+
+      camera = new THREE.PerspectiveCamera( 45, window.innerWidth / window.innerHeight, 0.1, 1000 );
+      camera.position.y = -1;
+
+      scene = new THREE.Scene(); 
+
+      renderer = new THREE.WebGLRenderer();
+      renderer.setPixelRatio( window.devicePixelRatio );
+      renderer.setSize(window.innerWidth, window.innerHeight);
+      document.body.appendChild(renderer.domElement);
+
+      const controls = new OrbitControls( camera, renderer.domElement );
+      controls.update();
+
+      stats = new Stats();
+      document.body.appendChild(stats.dom);
+      window.addEventListener("resize", onWindowResize);
+
+      let material = new THREE.ShaderMaterial({
+        uniforms: {
+          u_texture: { type: "t", value: null },
+          u_depth: { type: "t", value: null },
+        },
+        vertexShader: document.getElementById("shader-vs").textContent.trim(),
+        fragmentShader: document.getElementById("shader-fs").textContent.trim(),
+      });
+
+      let geometry = new THREE.BufferGeometry();
+      const vertices = new Float32Array([-1,-1,0,  1,-1,0,  1,1,0, -1,-1,0,  1,1,0,  -1,1,0]);
+      geometry.setAttribute(
+        "position",
+        new THREE.BufferAttribute(vertices, 3)
+      );
+      mesh = new THREE.Mesh(geometry, material);
+      mesh.position.set( -0.5, -0.5, 0 );
+      scene.add(mesh);
+
+      ///// test
+      const boxGeometry = new THREE.BoxGeometry( 0.25,  0.25,  0.25 );
+      const texture = new THREE.TextureLoader().load( 'crate.gif' );
+      let boxMaterial = new THREE.MeshBasicMaterial( { 
+        map: texture,
+        depthTest: true,
+        depthWrite: true,
+       } );
+      let boxObject = new THREE.Mesh( boxGeometry,  boxMaterial);
+      boxObject.position.set( 0, 0, 0 );
+      scene.add( boxObject );
+      /////
+
+      camera.lookAt(scene.position);
+      animate();
+
+      function animate() {
+        requestAnimationFrame(animate);
+        controls.update();
+		
+        let width = window.innerWidth;
+        let height = window.innerHeight;
+        if (udSDKJSReady) {
+          udSDKJS_ResizeScene(width, height, 0, 0);
+
+          var v = mesh.modelViewMatrix.elements;
+          udSDKJS_SetMatrix("view", v[0],v[1],v[2],v[3],v[4],v[5],v[6],v[7],v[8],v[9],v[10],v[11],v[12],v[13],v[14],v[15]);
+
+          var v = camera.projectionMatrix.elements;
+          udSDKJS_SetMatrix("projection", v[0],v[1],v[2],v[3],v[4],v[5],v[6],v[7],v[8],v[9],v[10],v[11],v[12],v[13],v[14],v[15]);
+
+          udSDKJS_RenderQueue();
+          var ptr = udSDKJS_GetColourBuffer();
+          var data = new Uint8Array(
+            HEAPU8.subarray(ptr, ptr + width * height * 4)
+          );
+
+          colourTextureGL = new THREE.DataTexture(
```

Comment:
Is it possible to create this (and the depth texture below) once (on resize), and then upload the pixels every frame?

Reply:
fixed.